### ISSUE TITLE
Improve status log

### ISSUE THREAD

**Issue by splix** [ethereumproject/go-ethereum#127]
Currently geth has non-informative log, displaying download status only every 2048 blocks, which can take several hours per log line for block range 2,4xx,xxx.

It's better to display status every 1 minute, with details similar to Parity's. Suggested log is:
```
2017-02-03 16:49:00  Sync   #3124227 of #3124363 c76c…34e7   77/ 242/ 7 blk/tx/mgas sec    1/ 4/25 peers
2017-02-03 16:50:00  Sync   #3124247 of #3124363 75e4…8eff   51/  51/ 5 blk/tx/mgas sec    1/ 4/25 peers
2017-02-03 16:51:00  Sync   #3124567 of #3124363 9af3…34ae  117/ 129/11 blk/tx/mgas sec    2/ 5/25 peers
2017-02-03 16:52:00  Sync   #3124787 of #3124363 1e3a…8351    9/   6/ 1 blk/tx/mgas sec    1/ 7/25 peers
2017-02-03 16:52:05  Import #3124788             84e1…1ff4        15/ 7 tx/mgas            3/10/25 peers
2017-02-03 16:52:25  Import #3124789             9e45…a241         5/ 1 tx/mgas            5/12/25 peers
2017-02-03 16:52:45  Import #3124790             d819…f71c         0/ 0 tx/mgas           11/18/25 peers
```

Where:

- `Fast`/`Sync` for initial sync, `Import` after that
- `#3124787 of #3124363` - block X of total height Y
- `c76c…34e7` block hash
- `77/ 242/ 7 blk/tx/mgas sec` performance for past minute, avg blocks/transactions/mgas processed per second. 3 character for block, 4 for transactions, 2 for mgas
- `1/ 4/25 peers` download from 1 peer, connected to 4, of max 25. 2 characters for each part

**Comment by whilei** [comment]
Also TODO: 

Use `glog` package to modularize geth output logging to centralize control over formatting, verbosity. As is assorted logging methods, ie fmt and log are used seemingly randomly.

**Comment by splix** [comment]
Now I think this task is bigger that we expected, let's move it to a next release?

**Comment by whilei** [comment]
Closing via #324

**Comment by splix** [comment]
1. should work by default, not because of special command line option. also, wasn't able to enable it on linux, works only on macos
2. prefix like `I1120 22:49:30.038282 cmd/geth/cmd.go:929]` doesn't make any sense, timestamp is enough
3. just `Sync` not `STATUS SYNC FullSync`
4. let's avoid `(49.76%)`, takes space for no reason
5. doesn't look like it shows progress, just stuck on same block
6. I see some weird trace-level logs in addition to status output. shouldn't be here

**Comment by whilei** [comment]
1. It works for me on linux:
```
freyabison@ribbon:~/geth $ ./geth --cache 5 --no-discover --max-peers=0 --log-status="sync=10" --verbosity 2
I1122 16:44:01.126103 cmd/geth/flag.go:421] WARNING: No etherbase set and no accounts found as default
I1122 16:44:01.132031 cmd/geth/flag.go:591] Machine logs: disabled
I1122 16:44:11.977780 cmd/geth/cmd.go:929] STATUS SYNC Discover #      0                     d4e…fa3    0/   0/ 0 blks/txs/mgas sec  0/ 0 peers
I1122 16:44:21.977749 cmd/geth/cmd.go:929] STATUS SYNC Discover #      0                     d4e…fa3    0/   0/ 0 blks/txs/mgas sec  0/ 0 peers
I1122 16:44:31.977801 cmd/geth/cmd.go:929] STATUS SYNC Discover #      0                     d4e…fa3    0/   0/ 0 blks/txs/mgas sec  0/ 0 peers
^CI1122 16:44:33.351044 cmd/geth/cmd.go:84] Got interrupt, shutting down...
freyabison@ribbon:~/geth $ geth version
-bash: geth: command not found
freyabison@ribbon:~/geth $ ./geth version
Geth
Version: v4.1.1+3-ca1f613
Protocol Versions: [63 62]
Network Id: 1
Go Version: go1.8
OS: linux
GOPATH=/home/freyabison/Go/imgopath
GOROOT=/home/freyabison/Go/imgroot/go
```

Maybe you can describe what you're seeing when not working for you on linux?

2. Rel #410
3. I used `STATUS SYNC FullSync` (which has alternate outputs as below):

```
STATUS SYNC FastSync
STATUS SYNC FullSync
STATUS SYNC Discover
```

because I anticipated that there may be other status logs that would eventually want to be included, like

```
STATUS MINE Mining
STATUS MINE Commit

STATUS STATE ValidateTx
STATUS STATE BroadcastTx

STATUS ACCOUNT Add
STATUS ACCOUNT Lock
```

4. I find it useful to easily understand rate of progress without pulling out my calculator.
5. It is based on a time intervals, not events, so if displayed statistics don't change within the interval it will appear static. 

For example:
```
~/Library/EthereumClassic  ⟠ geth --chain mlogfly --log-status="sync=10" --verbosity 2 --log-dir mlogfly/log --port=30305
I1123 07:58:55.722680 cmd/geth/flag.go:421] WARNING: No etherbase set and no accounts found as default
I1123 07:58:55.723727 cmd/geth/flag.go:591] Machine logs: disabled
I1123 07:59:08.046853 cmd/geth/cmd.go:929] STATUS SYNC Discover # 254193                     450…2dd    0/   0/ 0 blks/txs/mgas sec  1/25 peers
I1123 07:59:18.048980 cmd/geth/cmd.go:929] STATUS SYNC FullSync # 254208 of #4886528 (5.20%) cf6…ba6    1/   1/ 0 blks/txs/mgas sec  1/25 peers
I1123 07:59:28.163037 cmd/geth/cmd.go:929] STATUS SYNC FullSync # 255018 of #4886528 (5.22%) 008…93f   81/ 156/ 4 blks/txs/mgas sec  1/25 peers
I1123 07:59:38.266593 cmd/geth/cmd.go:929] STATUS SYNC FullSync # 256937 of #4886528 (5.26%) ab9…e35  191/ 164/ 3 blks/txs/mgas sec  1/25 peers
I1123 07:59:48.077975 cmd/geth/cmd.go:929] STATUS SYNC FullSync # 257649 of #4886528 (5.27%) 5be…218   71/  80/ 1 blks/txs/mgas sec  2/25 peers
I1123 07:59:58.041863 cmd/geth/cmd.go:929] STATUS SYNC FullSync # 257649 of #4886528 (5.27%) 5be…218    0/   0/ 0 blks/txs/mgas sec  2/25 peers
I1123 08:00:08.042088 cmd/geth/cmd.go:929] STATUS SYNC FullSync # 257652 of #4886528 (5.27%) 803…c0b    0/   0/ 0 blks/txs/mgas sec  2/25 peers
I1123 08:00:18.346671 cmd/geth/cmd.go:929] STATUS SYNC FullSync # 259197 of #4886528 (5.30%) df6…d55  154/ 114/ 6 blks/txs/mgas sec  2/25 peers
I1123 08:00:28.320982 cmd/geth/cmd.go:929] STATUS SYNC FullSync # 260582 of #4886528 (5.33%) bf3…c79  138/ 196/ 4 blks/txs/mgas sec  2/25 peers
I1123 08:00:38.300362 cmd/geth/cmd.go:929] STATUS SYNC FullSync # 262113 of #4886528 (5.36%) 176…ef5  153/ 116/ 2 blks/txs/mgas sec  2/25 peers
```

Perhaps this may be an opportunity to think about other `STATUS` log topics, as above, which would show progress if not necessarily block import/sync progress.

6. You can adjust the general verbosity with `--verbosity=n`. `--log-status` currently uses `Error` verbosity, so
in order to only view status logs and errors (which I expect anyone would want to see), you can use

```
--log-status="sync=10" --verbosity 1
```

----

One possible simple solve for your complaints would be to change the default settings for `--verbosity`, `--vmodule`, and `--log-status`. For example:

```
~/Library/EthereumClassic  ⟠ geth --log-status="sync=5" --verbosity 1 --vmodule="cmd/geth/*=3"
I1123 08:07:02.824060 cmd/geth/flag.go:816] --------------------------------------------------------------------------------------------------------------
I1123 08:07:02.824085 cmd/geth/flag.go:818] Starting Geth Classic source.v4.1.1+20-d38a52e.mlog-block-difficulty.17-11-21-16-53-51
I1123 08:07:02.824103 cmd/geth/flag.go:819] Geth is configured to use ETC blockchain: Ethereum Classic Mainnet
I1123 08:07:02.824129 cmd/geth/flag.go:820] Using chain database at: /Users/ia/Library/EthereumClassic/mlogfly/chaindata
I1123 08:07:02.824143 cmd/geth/flag.go:822] 5 blockchain upgrades associated with this configuration:
I1123 08:07:02.824158 cmd/geth/flag.go:825]  1150000 Homestead
I1123 08:07:02.824196 cmd/geth/flag.go:825]  1920000 The DAO Hard Fork
I1123 08:07:02.824214 cmd/geth/flag.go:827]          with block 0x94365e3a8c0b35089c1d1195081fe7489b528a84b22199c916180db8b28ade7f
I1123 08:07:02.824223 cmd/geth/flag.go:825]  2500000 GasReprice
I1123 08:07:02.824235 cmd/geth/flag.go:825]  3000000 Diehard
I1123 08:07:02.824251 cmd/geth/flag.go:825]  5000000 Gotham
I1123 08:07:02.824276 cmd/geth/flag.go:838] State starting nonce: 0
I1123 08:07:02.824293 cmd/geth/flag.go:841] --------------------------------------------------------------------------------------------------------------
I1123 08:07:02.824505 cmd/geth/flag.go:421] WARNING: No etherbase set and no accounts found as default
I1123 08:07:02.824965 cmd/geth/flag.go:591] Machine logs: disabled
I1123 08:07:04.937583 cmd/geth/cmd.go:824] STATUS SYNC Log interval set: 5 seconds
I1123 08:07:09.939686 cmd/geth/cmd.go:929] STATUS SYNC Discover # 263107                     fcc…0b2    0/   0/ 0 blks/txs/mgas sec  0/25 peers
I1123 08:07:14.938607 cmd/geth/cmd.go:929] STATUS SYNC Discover # 263107                     fcc…0b2    0/   0/ 0 blks/txs/mgas sec  0/25 peers
I1123 08:07:19.942139 cmd/geth/cmd.go:929] STATUS SYNC Discover # 263107                     fcc…0b2    0/   0/ 0 blks/txs/mgas sec  1/25 peers
I1123 08:07:24.938110 cmd/geth/cmd.go:929] STATUS SYNC Discover # 263107                     fcc…0b2    0/   0/ 0 blks/txs/mgas sec  1/25 peers
I1123 08:07:29.938100 cmd/geth/cmd.go:929] STATUS SYNC Discover # 263107                     fcc…0b2    0/   0/ 0 blks/txs/mgas sec  1/25 peers
I1123 08:07:34.936578 cmd/geth/cmd.go:929] STATUS SYNC Discover # 263107                     fcc…0b2    0/   0/ 0 blks/txs/mgas sec  1/25 peers
I1123 08:07:39.937234 cmd/geth/cmd.go:929] STATUS SYNC Discover # 263107                     fcc…0b2    0/   0/ 0 blks/txs/mgas sec  3/25 peers
I1123 08:07:44.935521 cmd/geth/cmd.go:929] STATUS SYNC Discover # 263107                     fcc…0b2    0/   0/ 0 blks/txs/mgas sec  3/25 peers
I1123 08:07:49.936484 cmd/geth/cmd.go:929] STATUS SYNC Discover # 263107                     fcc…0b2    0/   0/ 0 blks/txs/mgas sec  3/25 peers
I1123 08:07:54.935826 cmd/geth/cmd.go:929] STATUS SYNC Discover # 263107                     fcc…0b2    0/   0/ 0 blks/txs/mgas sec  3/25 peers
I1123 08:07:59.936243 cmd/geth/cmd.go:929] STATUS SYNC Discover # 263107                     fcc…0b2    0/   0/ 0 blks/txs/mgas sec  3/25 peers
I1123 08:08:04.934568 cmd/geth/cmd.go:929] STATUS SYNC Discover # 263107                     fcc…0b2    0/   0/ 0 blks/txs/mgas sec  3/25 peers
I1123 08:08:09.937045 cmd/geth/cmd.go:929] STATUS SYNC Discover # 263107                     fcc…0b2    0/   0/ 0 blks/txs/mgas sec  3/25 peers
I1123 08:08:14.934633 cmd/geth/cmd.go:929] STATUS SYNC Discover # 263107                     fcc…0b2    0/   0/ 0 blks/txs/mgas sec  3/25 peers
I1123 08:08:19.935809 cmd/geth/cmd.go:929] STATUS SYNC Discover # 263107                     fcc…0b2    0/   0/ 0 blks/txs/mgas sec  3/25 peers
I1123 08:08:24.932957 cmd/geth/cmd.go:929] STATUS SYNC Discover # 263107                     fcc…0b2    0/   0/ 0 blks/txs/mgas sec  4/25 peers
```

**Comment by splix** [comment]
1. just nothing, looks like it stuck. tried few times, blank screen, or error logs if try to emulate network problems. so geth works, but doesn't give status feedback log
3. there will be _only_ sync and import in that status. it's not logs
4. but it will always be 98% or 99%, except initial import, why? and if you need it, time in hours makes more sense during initial import, rather than percentage
5. ah, I see, it was on `STATUS SYNC Discover` mode. I didn't plan Discover mode initially, only sync, so didn't realize it's not a sync. let's discuss it on separate ticket do we need discover feedback and if we need how it should look like (because _current block num_ and _blks/txs/mgas sec_ are definitely meaningless in that context)
6. let's make good default output. users will never dig into command line options for that, they will quit instead

**Comment by splix** [comment]
Updated
```
2017-02-03 16:44:00  Discover                                                              0/25 peers
2017-02-03 16:45:00  Discover                                                              1/25 peers
2017-02-03 16:46:00  Fast   #2481951 of #3124363    79.4%   1211/  554 blk  / mgas sec     6/25 peers
2017-02-03 16:47:00  Fast   #2689911 of #3124363    86.1%    611/  981 blk  / mgas sec     6/25 peers
2017-02-03 16:48:00  Fast   #2875913 of #3124363    92.0%    502/  760 blk  / mgas sec     4/25 peers
2017-02-03 16:49:00  Sync   #3124227 of #3124363 c76c34e7   77/ 242/ 7 blk/tx/mgas sec     4/25 peers
2017-02-03 16:50:00  Sync   #3124247 of #3124363 75e48eff   51/  51/ 5 blk/tx/mgas sec     4/25 peers
2017-02-03 16:51:00  Sync   #3124567 of #3124363 9af334ae  117/ 129/11 blk/tx/mgas sec     5/25 peers
2017-02-03 16:52:00  Sync   #3124787 of #3124363 1e3a8351    9/   6/ 1 blk/tx/mgas sec     7/25 peers
2017-02-03 16:52:05  Import #3124788             84e11ff4        15/ 7 tx/mgas            10/25 peers
2017-02-03 16:52:25  Import #3124789             9e45a241         5/ 1 tx/mgas            12/25 peers
2017-02-03 16:52:45  Import #3124790             d819f71c         0/ 0 tx/mgas            18/25 peers
```

Where:

- `Fast`/`Sync` for initial sync, `Import` after that, `Discover` on start
- `#3124787 of #3124363` - block X of total height Y
- `c76634e7` block hash
- `77/ 242/ 7 blk/tx/mgas sec` performance for past minute, avg blocks/transactions/mgas processed per second. 3 character for block, 4 for transactions, 2 for mgas
- `4/25 peers` connected to 4, of max 25. 2 characters for each part

**Comment by whilei** [comment]
So for all modes besides `Import` it should print status logs only every 1 minute, but for `Import` it should print a line for each block it imports (not time-interval based)?

**Comment by whilei** [comment]
And are all error and warning logs entirely suppressed?

**Comment by splix** [comment]
Yes, for Import it should print it at the moment of import. Other statuses just aggregation, one per minute.

Errors/warning should be displayed in normal way, but we may want to reconsider what should be logged as error/warning. It's a separate issue

**Comment by whilei** [comment]
Closed with #423
